client/src/components/Form/Elements/FormText.vue

```diff
@@ -0,0 +1,144 @@
+                 -->
+
+                <component
+                    :is="componentName"
+                    :id="id"
+                    :readonly="readonly"
+                    :placeholder="placeholder"
+                    :style="style"
+                    :type="type"
+                    v-model="currentValue"
+                    @change="onInputChange" />
+
+                <datalist v-if="datalist && !area && !multiple" :id="`${id}-datalist`">
+                    <option v-for="d in datalist" value="d"></option>
+                </datalist>
+            </b-col>
+        </b-row>
+    </div>
+</template>
+
+<script>
+export default {
+    props: {
+        value: {
+            // String or Array for multiple
+            required: false,
+            default: "",
+        },
+        id: {
+            type: String,
+            required: false,
+        },
+        type: {
+            type: String,
+            required: false,
+            default: "text",
+            validator: (prop) => ["text", "password"].includes(prop.toLowerCase()),
+        },
+        area: {
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        multiple: {
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        readonly: {
+            type: Boolean,
+            required: false,
+            default: false,
+        },
+        placeholder: {
+            type: String,
+            required: false,
+        },
+        color: {
+            type: String,
+            required: false,
+        },
+        styleObj: {
+            type: Object,
+            required: false,
+            default: () => {},
+        },
+        datalist: {
+            // Display list of suggestions in autocomplete dialog
+            type: Array,
+            required: false,
+        },
+    },
+    data() {
+        return {
+            dismissSecs: 5,
+            dismissCountDown: 0,
+            errorMessage: "",
+        };
+    },
+    computed: {
+        currentValue: {
+            get() {
+                // TODO: is silent fail on non-strings appropriate?
+                const v = this.value || "";
+                if (typeof v === "array") {
+                    return this.multiple ? v.map((i) => String(i)).join("\n") : String(v[0]);
+                }
+                return typeof v === "string" ? v : "";
+            },
+            // TODO: handle the emmitted "input"
+            set(newVal, oldVal) {
+                if (newVal !== oldVal) {
+                    this.$emit("input", newVal);
+                }
+            },
+        },
+        componentName() {
+            return this.area || this.multiple ? "b-form-textarea" : "b-form-input";
+        },
+        style() {
+            return this.color ? { ...this.styleObj, color: this.color } : this.styleObj;
+        },
+    },
+    methods: {
+        onInputChange(value) {
```

Comment:
Validation happens right now `validateInputs` (see: `Form/utilities.js`) for all inputs together. I think once we have transformed all inputs to Vue, we should add concept to (additionally?) validate on the input component level but for now this is not necessary.